src/Command/TransferOwnershipCommand.php

```diff
@@ -0,0 +1,172 @@
+
+        if (!$dryRun) {
+            $this->transferOwnership($packages, $maintainers);
+        }
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * @return User[]
+     */
+    private function queryAndValidateMaintainers(InputInterface $input, OutputInterface $output): array
+    {
+        $usernames = array_map('strtolower', $input->getArgument('maintainers'));
+        sort($usernames);
+
+        $maintainers = $this->getEM()->getRepository(User::class)->findUsersByUsername($usernames, ['usernameCanonical' => 'ASC']);
+
+        if (array_keys($maintainers) === $usernames) {
+            return $maintainers;
+        }
+
+        $notFound = [];
+
+        foreach ($usernames as $username) {
+            if (!array_key_exists($username, $maintainers)) {
+                $notFound[] = $username;
+            }
+        }
+
+        sort($notFound);
+
+        $output->writeln(sprintf('<error>%d maintainers could not be found: %s</error>', count($notFound), implode(', ', $notFound)));
+
+        return [];
+    }
+
+    /**
+     * @return Package[]
+     */
+    private function queryVendorPackages(string $vendor): array
+    {
+        return $this->getEM()
+            ->getRepository(Package::class)
+            ->getFilteredQueryBuilder(['vendor' => $vendor], true)
+            ->getQuery()
+            ->getResult();
+    }
+
+    /**
+     * @param Package[] $packages
+     * @param User[] $maintainers
+     */
+    private function outputPackageTable(OutputInterface $output, array $packages, array $maintainers): void
+    {
+        $rows = [];
+
+        $newMaintainers = array_map(fn (User $user) => $user->getUsername(), $maintainers);
+
+        foreach ($packages as $package) {
+            $currentMaintainers = $package->getMaintainers()->map(fn (User $user) => $user->getUsername())->toArray();
+            sort($currentMaintainers);
+
+            $rows[] = [
+                $package->getVendor(),
+                $package->getPackageName(),
+                implode(', ', $currentMaintainers),
+                implode(', ', $newMaintainers),
+            ];
+        }
+
+        $table = new Table($output);
+        $table
+            ->setHeaders(['Vendor', 'Package', 'Current Maintainers', 'New Maintainers'])
+            ->setRows($rows)
+        ;
+        $table->render();
+    }
+
+    /**
+     * @param Package[] $packages
+     * @param User[] $maintainers
+     */
+    private function transferOwnership(array $packages, array $maintainers): void
+    {
+        foreach ($packages as $package) {
+            $oldMaintainers = $package->getMaintainers()->toArray();
+
+            $package->getMaintainers()->clear();
+            foreach ($maintainers as $maintainer) {
+                $package->addMaintainer($maintainer);
+            }
+
+            $this->doctrine->getManager()->persist(AuditRecord::packageTransferred($package, null, $oldMaintainers, array_values($maintainers)));
```

Comment:
What about packages that were already own by the new maintainers ? Should they actually be marked as transferred to the same maintainers ?

Reply:
Yeah I think what we could do here is at least check that the old and new arent' an exact match and if they are skip transferring that package as that's a noop. I'll add that post merge.